lib/lotus/commands/new/abstract.rb

```diff
@@ -0,0 +1,204 @@
+require 'lotus/utils/hash'
+module Lotus
+  module Commands
+    class New < Thor
+      class Abstract
```

Comment:
Yeahhh. Kind of got huge that PR. One thing lead to another. Not sure there is an easy way to split it up though because it would require to fix a lot of tests that get deleted anyway. Any suggestions?

I think that in total this PR improves the generators but it's still far from perfect. I stopped at some point in order to get feedback ( @joneslee85 suggested to apply the changes to all generators to get to a somewhat consistent state).